test/engine-luatest/pagination_test.lua

```diff
@@ -51,14 +58,30 @@ tree_g.after_each(function(cg)
 end)
 
 tree_g.test_tree_pagination = function(cg)
-    cg.server:exec(function()
-        local s = box.space.s
-        s:create_index("pk", {type = "tree"})
-        s:create_index("sk", {parts = {2}, type = "tree", unique=false})
-        s:create_index("sk2", {parts = {3}, type = "tree", unique=false})
-        s:create_index("sk_unique", {parts = {4}, type = "tree", unique=true})
+    cg.server:exec(function(is_rv)
+        box.space.s:create_index("pk", {type = "tree"})
+        box.space.s:create_index("sk", {
+            parts = {2},
+            type = "tree",
+            unique = false
+        })
+        box.space.s:create_index("sk2", {
+            parts = {3},
+            type = "tree",
+            unique = false
+        })
+        box.space.s:create_index("sk_unique", {
+            parts = {4},
+            type = "tree",
+            unique = true
+        })
 
         -- Fetch position in empty space
+        local s = box.space.s
+        if is_rv then
+            local rv = box.read_view.open()
+            s = rv.space.s
+        end
```

Comment:
Can you move all the initialization to `before_test` callback and then replace the space with a read view in a common `before_each` callback?